adapters/repos/db/vector/flat/index.go

```diff
@@ -0,0 +1,524 @@
+	}
+	vec, err := index.bq.Encode(vector)
+	if err != nil {
+		return err
+	}
+	slice := make([]byte, len(vec)*8)
+	index.storeCompressedVector(id, byteSliceFromUint64Slice(vec, slice))
+
+	return nil
+}
+
+func (index *flat) Delete(ids ...uint64) error {
+	return nil
+}
+
+func (index *flat) searchTimeEF(k int) int {
+	if index.compression == flatent.CompressionNone {
+		return k
+	}
+	// load atomically, so we can get away with concurrent updates of the
+	// userconfig without having to set a lock each time we try to read - which
+	// can be so common that it would cause considerable overhead
+	ef := int(atomic.LoadInt64(&index.ef))
+	if ef < k {
+		ef = k
+	}
+	return ef
+}
+
+func (index *flat) SearchByVector(vector []float32, k int, allow helpers.AllowList) ([]uint64, []float32, error) {
+	if index.distancerProvider.Type() == "cosine-dot" {
+		// cosine-dot requires normalized vectors, as the dot product and cosine
+		// similarity are only identical if the vector is normalized
+		vector = distancer.Normalize(vector)
+	}
+
+	ef := index.searchTimeEF(k)
+	heap := index.pqResults.GetMax(ef)
+	defer index.pqResults.Put(heap)
+
+	firstId := uint64(0)
+	alreadyFound := 0
+	if allow != nil {
+		firstId, _ = allow.Iterator().Next()
+	}
+
+	if index.compression != flatent.CompressionNone {
+		query, err := index.bq.Encode(vector)
+		if err != nil {
+			return nil, nil, err
+		}
+		cursor := index.store.Bucket(helpers.CompressedObjectsBucketLSM).Cursor()
+		var key []byte
+		var v []byte
+		if allow != nil {
+			buff := make([]byte, 16)
+			binary.BigEndian.PutUint64(buff[8:], firstId)
+			key, v = cursor.Seek(buff)
+		} else {
+			key, v = cursor.First()
+		}
+		for key != nil && (allow == nil || alreadyFound < allow.Len()) {
+			alreadyFound++
+			id := binary.LittleEndian.Uint64(key)
+			if allow != nil && !allow.Contains(id) {
+				continue
+			}
+			t := index.pool.uint64SlicePool.Get(len(v) / 8)
+			candidate := uint64SliceFromByteSlice(v, t.slice)
+			d, _ := index.bq.DistanceBetweenCompressedVectors(candidate, query)
+
+			index.pool.uint64SlicePool.Put(t)
+			if heap.Len() < ef || heap.Top().Dist > d {
+				if heap.Len() == ef {
+					heap.Pop()
+				}
+				heap.Insert(id, d)
+			}
+			key, v = cursor.Next()
+		}
+		cursor.Close()
+
+		ids := make([]uint64, ef)
+		for j := range ids {
+			ids[j] = heap.Pop().ID
+		}
+
+		for _, id := range ids {
+			candidate, err := index.vectorForID(context.Background(), id)
+			if err != nil {
+				return nil, nil, err
+			}
+			d, _, _ := index.distancerProvider.SingleDist(candidate, vector)
+			if heap.Len() < ef || heap.Top().Dist > d {
+				if heap.Len() == ef {
+					heap.Pop()
+				}
+				heap.Insert(uint64(id), d)
+			}
+		}
+		for heap.Len() > k {
+			heap.Pop()
+		}
+	} else {
+		cursor := index.flatStore()
+		var key []byte
+		var v []byte
+		if allow != nil {
+			buff := make([]byte, 16)
+			binary.BigEndian.PutUint64(buff[8:], firstId)
+			key, v = cursor.Seek(buff)
```

Comment:
I need to verify that, but I'm afraid keys in objects bucket are uuids, therefore Seeking for uint64 does not make sense - nothing will be returned